### PR TITLE
feat(hub-discussions): add channelAcl to canCreateChannel

### PR DESCRIPTION
**Pull request by velveetachef**
1. Description: refactor `canCreateChannel` to check channel permissions in the `channelAcl` property, if included.
  - rename a few `.spec` test file to `.test` so jasmine runner picks them up
  - refactor legacy channel permission checks
  - remove unused parameter from `intersectGroups` for test coverage

1. Instructions for testing:

1. Closes Issues: #<number> (if appropriate)

1. [x] used semantic commit messages
  
1. [ ] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

**Comment by rgwozdz** [comment]
@velveetachef - note that the PR title has to be in the form of a semanitc commit message:

> PR title follows semantic commit format (CRITICAL if the title is not in a semantic format, the release automation will not run!)

File: packages/discussions/src/utils/channels/can-create-channel.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+function canAllowGroups(
+  user: IDiscussionsUser,
+  groupPermissions?: IChannelAclPermissionDefinition[]
+) {
+  if (!groupPermissions) {
+    return true;
+  }
+  return isMemberOfAllChannelGroups(user.groups, groupPermissions);
```
Comment (rgwozdz) [inline packages/discussions/src/utils/channels/can-create-channel.ts:93]: membership check is good, but you also have to check the group's typeKeyword array for existence of `cannotDiscuss`.  See https://devtopia.esri.com/dc/hub-discussions/blob/main/src/common/utils/validate-all-groups-can-discuss.ts#L52-L60.

**Review by rgwozdz — APPROVED** [review]
Just the one comment about `cannotDiscuss`.

**Review by rgwozdz — CHANGES_REQUESTED** [review]
Oops. I meant to do "request changes".

File: packages/discussions/src/utils/channels/can-create-channel.ts (hunk 1 above)
Comment (velveetachef) [reply packages/discussions/src/utils/channels/can-create-channel.ts]: 👍

**Review by rgwozdz — APPROVED** [review]

**Pull request merged**